rllib/agents/trainer.py

```diff
@@ -653,50 +653,65 @@ def default_logger_creator(config):
 
     @override(Trainable)
     def setup(self, config: PartialTrainerConfigDict):
-        env = self._env_id
-        if env:
-            config["env"] = env
+
+        # Setup our config: Merge the user-supplied config (which could
+        # be a partial config dict with the class' default).
+        self.config = self.merge_trainer_configs(
+            self.get_default_config(), config, self._allow_unknown_configs)
+
+        # Setup the "env creator" callable.
+        if self._env_id:
+            self.config["env"] = self._env_id
+
             # An already registered env.
-            if _global_registry.contains(ENV_CREATOR, env):
-                self.env_creator = _global_registry.get(ENV_CREATOR, env)
-            # A class specifier.
-            elif "." in env:
+            if _global_registry.contains(ENV_CREATOR, self._env_id):
```

Comment:
maybe worth breaking everything in this if statement into a little util function, so things don't look so nested here.